pkg/controller/master/setting/load_from_config.go

```diff
@@ -0,0 +1,37 @@
+package setting
+
+import (
+	"github.com/sirupsen/logrus"
+
+	harvesterv1 "github.com/harvester/harvester/pkg/apis/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/generated/controllers/harvesterhci.io/v1beta1"
+	"github.com/harvester/harvester/pkg/util"
+)
+
+type LoadingPreconfigHandler struct {
+	namespace string
+	settings  v1beta1.SettingClient
+}
+
+func (h *LoadingPreconfigHandler) settingOnChanged(_ string, setting *harvesterv1.Setting) (*harvesterv1.Setting, error) {
+	if setting == nil || setting.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	settingCpy := setting.DeepCopy()
```

Comment:
Move this into the following if block so that we only do deep copy when the update is needed.

Reply:
Moved into the if block.